--resource-dir needs to be made absolute, similar to other path flags like --sysroot or -I for cgo

### What version of rules_go are you using?

master @ 24c3d41068c57d1ee7bf5774391285fd4ac5d189

### What version of gazelle are you using?

v0.24

### What version of Bazel are you using?

4.10

### Does this issue reproduce with the latest releases of all the above?

yes

### What operating system and processor architecture are you using?

mac/linux x86_64

### Any other potentially useful information about your toolchain?

custom c++ clang based toolchain with `--resource-dir` set to control location of compiler files

### What did you do?

build go code with cgo enabled with my custom toolchain

### What did you expect to see?

a successful build 

### What did you see instead?

```
 (cd /Users/chancila/<REDACTED> && \
  exec env - \
    CC=external/<REDACTED>/llvm/bin/clang \
    CGO_CFLAGS='--sysroot=external/<REDACTED>/sysroots/macos_12_5_1 --target=arm64-apple-macos11.0 -resource-dir external/<REDACTED>/llvm/lib/clang/12.0.0 -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__="redacted" -D__TIMESTAMP__="redacted" -D__TIME__="redacted" -fdebug-compilation-dir .' \
    CGO_ENABLED=1 \
    CGO_LDFLAGS='--sysroot=external/<REDACTED>/sysroots/macos_12_5_1 --target=arm64-apple-macos11.0 -fuse-ld=lld -mlinker-version=519 -Wl,-sdk_version,10.14 -fuse-ld=/usr/bin/ld' \
    GOARCH=arm64 \
    GOOS=darwin \
    GOPATH='' \
    GOROOT=external/go_sdk \
    GOROOT_FINAL=GOROOT \
    PATH=external/<REDACTED>/llvm/bin:/bin:/usr/bin \
  bazel-out/host/bin/external/go_sdk/builder stdlib -sdk external/go_sdk -installsuffix darwin_arm64 -out bazel-out/darwin-fastbuild/bin/external/io_bazel_rules_go/stdlib_)

# plugin
In file included from bazel-out/darwin-fastbuild/bin/external/io_bazel_rules_go/stdlib_/src/plugin/plugin_dlopen.go:11:
external/<REDACTED>/sysroots/macos_12_5_1/usr/include/dlfcn.h:40:10: fatal error: 'stdbool.h' file not found
#include <stdbool.h>
         ^~~~~~~~~~~
```

Note the failure to find stdbool.h, stdbool is a compiler provided file and in clang the inclusion of the compiler built-ins can be controled via `--resource-dir`, I was able to trace the clang invocations and noticed that the `--resource-dir` flag was relative and pointing to an invalid path, causing compiler built-ins to not be included while the other flags like --sysroot and the inclusion flags were made absolute, it seems the builder code does not account for this flag.

--- a/go/tools/builders/env.go
+++ b/go/tools/builders/env.go
@@ -34,7 +34,7 @@ var (
 	// cgoEnvVars is the list of all cgo environment variable
 	cgoEnvVars = []string{"CGO_CFLAGS", "CGO_CXXFLAGS", "CGO_CPPFLAGS", "CGO_LDFLAGS"}
 	// cgoAbsEnvFlags are all the flags that need absolute path in cgoEnvVars
-	cgoAbsEnvFlags = []string{"-I", "-L", "-isysroot", "-isystem", "-iquote", "-include", "-gcc-toolchain", "--sysroot"}
+	cgoAbsEnvFlags = []string{"-I", "-L", "-isysroot", "-isystem", "-iquote", "-include", "-gcc-toolchain", "--sysroot", "-resource-dir"}
 )
 
 // env holds a small amount of Go environment and toolchain information

